dask_expr/io/json.py

```diff
@@ -172,7 +170,7 @@ def to_json(
     from dask.dataframe.io.json import to_json
 
     return to_json(
-        df.to_legacy_dataframe(),
+        df,
```

Comment:
Over in `orc.py` this is `df.optimize()` -- should we do something similar here? Or is the `.optimize()` maybe not needed over there?

Reply:
This calls to_delayed, which triggers the optimization. I'd rather collect them in a single place